content/references/http-websocket-apis/public-api-methods/ledger-methods/ledger_data.md

```diff
@@ -251,6 +251,8 @@ The response follows the [standard format][], with a successful result containin
 | `state`        | Array                               | Array of JSON objects containing data from the ledger's state tree, as defined below. |
 | `marker`       | [Marker][]                          | Server-defined value indicating the response is paginated. Pass this to the next call to resume where this call left off. |
 
+If a `type` field is mentioned in the request, the `state` array might be empty. In such cases, you'll have to use the `marker` from this response to paginate and retrieve further data. Keep repeating this until you get a non-empty `state` array _(if the object specified in `type` exists on the ledger)_.
```

Comment:
Just reading this explanation, I don't feel like it provides a clear understanding what is going on under the hood vs. just having a usage suggestion

```suggestion
If a `type` field is mentioned in the request, the `state` array may be empty if the first set of array of objects does not match the type requested. In such cases, you may want to use the `marker` from this response to paginate and retrieve further data.
```

Reply:
Consider replacing a _may_ with _might_ to avoid ambiguity:
If a `type` field is mentioned in the request, the `state` array might be empty if the first set of array of objects does not match the type requested. In such cases, you can use the `marker` from this response to paginate and retrieve further data.

Reply:
@kennyzlei @DennisDawson why is it important for whoever is using the API to know what's going on underneath the hood? I thought we would like to hide such implementation detail for the sake of abstraction

Reply:
I think upon reading the usage suggestion sentence you have, my first thought is "Why do I need to keep paginating?  Is this a bug and this is a hacky workaround?  Seems weird to me".  If I knew how pagination comes first and then filtering on type, I understand why the response comes out a certain way, usually you assume the opposite as in filter then paginate if necessary

Reply:
cool that make sense if exposing a slight implementation detail gives more clarity to the user. It normally would make sense to do filter then paginate, but that can't be done with the ledger. If this is implemented, that would mean some one could try to call the api on a ledger index that doesn't have this object type, which would last hours to search through millions of ledger objects

Reply:
@DennisDawson thanks Dennis, updated